ops/deployer/gauntlet/gauntlet.go

```diff
@@ -0,0 +1,534 @@
+			},
+			ConfigEncryptionPublicKey: types.ConfigEncryptionPublicKey(configPKByte),
+		})
+	}
+
+	status := utils.LogStatus("InitOCR: set config test args")
+	alphaPPB := uint64(1000000)
+	_, _, f, onchainConfig, offchainConfigVersion, _, err := confighelper.ContractSetConfigArgsForTests(
+		2*time.Second,        // deltaProgress time.Duration,
+		5*time.Second,        // deltaResend time.Duration,
+		1*time.Second,        // deltaRound time.Duration,
+		500*time.Millisecond, // deltaGrace time.Duration,
+		10*time.Second,       // deltaStage time.Duration,
+		3,                    // rMax uint8,
+		S,                    // s []int,
+		helperOracles,        // oracles []OracleIdentityExtra,
+		median.OffchainConfig{
+			false,
+			alphaPPB,
+			false,
+			alphaPPB,
+			15 * time.Second,
+		}.Encode(), //reportingPluginConfig []byte,
+		500*time.Millisecond, // maxDurationQuery time.Duration,
+		500*time.Millisecond, // maxDurationObservation time.Duration,
+		500*time.Millisecond, // maxDurationReport time.Duration,
+		2*time.Second,        // maxDurationShouldAcceptFinalizedReport time.Duration,
+		2*time.Second,        // maxDurationShouldTransmitAcceptedReport time.Duration,
+		1,                    // f int,
+		[]byte{},             // onchainConfig []byte (calculated by the contract)
+	)
+	if status.Check(err) != nil {
+		return err
+	}
+
+	status = utils.LogStatus("InitOCR: begin proposal")
+	err = t.gauntlet.ExecCommand(
+		"ocr2:begin_proposal",
+		t.gauntlet.Flag("network", t.network),
+		t.Account[OCR2],
+	)
+
+	report, err := t.gauntlet.ReadCommandReport()
+	if err != nil {
+		return err
+	}
+	if err == nil && len(report.Data) == 0 {
+		err = errors.New("begin proposal produced no logs")
+	}
+
+	if status.Check(err) != nil {
+		return err
+	}
+
+	fmt.Printf(" - %s", report.Data["proposalId"])
+
+	var id string = report.Data["proposalId"]
+
+	// Be prepared to clear the proposal if incomplete.
+	defer func() {
+		if rerr == nil {
+			return // Success
+		}
+		// Failure: Try to clean up incomplete proposal.
+		status = utils.LogStatus("InitOCR: clear proposal: " + id)
+		err = t.gauntlet.ExecCommand(
+			"ocr2:clear_proposal",
+			t.gauntlet.Flag("network", t.network),
+			t.gauntlet.Flag("proposalId", id),
+			t.Account[OCR2],
+		)
+		if status.Check(err) != nil {
+			fmt.Println(err)
+			return
+		}
+	}()
+
+	jsonInput, err := json.Marshal(ProposeConfigDetails{
+		ID:            id,
+		Payees:        transmitterArray,
+		Transmitters:  transmitterArray,
+		F:             f,
+		OnchainConfig: onchainConfig,
+		Signers:       signersArray,
+	})
+	if err != nil {
+		return err
+	}
+
+	status = utils.LogStatus("InitOCR: propose config " + id)
+	err = t.gauntlet.ExecCommand(
+		"ocr2:propose_config",
+		t.gauntlet.Flag("network", t.network),
+		t.gauntlet.Flag("input", string(jsonInput)),
+		t.Account[OCR2],
+	)
+	if status.Check(err) != nil {
+		return err
+	}
+
+	offchainConfig := OffchainConfigDetails{
```

Comment:
The off-chain config is now defined twice in the same file, a few lines apart. I think we should be able to use the output of `confighelper.ContractSetConfigArgsForTests`.